### PR TITLE
NEXT-30813 - Checkout gateway

### PR DESCRIPTION
**Pull request by lernhart**

### CODE REVIEW

File: guides/plugins/apps/gateways/checkout/checkout-gateway.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+The Checkout Gateway aims to allow a streamlined implementation for making informed decisions during the checkout process, based on both the cart contents and the current sales channel context.
+In particular, the app system benefits from this solution, enabling seamless communication and decision-making on the app server during the checkout.
+
+While this documentation focuses on the app integration of the Checkout Gateway, the design is intended to allow a custom replacement solution via the plugin system."
+
+## Prerequisites
+
+You should be familiar with the concept of Apps, their registration flow as well as signing and verifying requests and responses between Shopware and the App backend server.
+
+{% page-ref page="app-base-guide.md" %}
```
Comment (bojanrajh) [inline guides/plugins/apps/gateways/checkout/checkout-gateway.md:16]: Use `<PageRef />` component instead ([example](https://github.com/shopware/docs/blob/3049f905a176b861b3c135490e46885e0a21f088/guides/plugins/index.md?plain=1#L16)).

File: guides/plugins/apps/gateways/checkout/checkout-gateway.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+Your app server must be also accessible for the Shopware server.
+You can use a tunneling service like [ngrok](https://ngrok.com/) for development.
+
+## Manifest configuration
+To indicate to Shopware that your app uses the checkout gateway, you must provide a `checkout` property inside a `gateways` parent property of your app's `manifest.xml`.
+
+Below, you can see an example definition of a working checkout gateway configuration.
+
+{% code title="manifest.xml" %}
```
Comment (bojanrajh) [inline guides/plugins/apps/gateways/checkout/checkout-gateway.md:26]: Use `::: code-group` block instead ([example](https://github.com/shopware/docs/blob/3049f905a176b861b3c135490e46885e0a21f088/guides/plugins/plugins/plugin-fundamentals/logging.md?plain=1#L62), [docs](https://vitepress.dev/guide/markdown#code-groups)).

File: guides/plugins/apps/gateways/checkout/checkout-gateway.md [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+After successful installation of your app, the checkout gateway will already be used during checkout.
+
+{% endcode %}
+
+## Checkout gateway endpoint
+During checkout, Shopware checks for any active checkout gateways and will call the `checkout` url.
+The app server will receive the current `SalesChannelContext`, `Cart`, and available payment and shipping methods as part of the payload.
+
+{% hint style="warning" %}
```
Comment (bojanrajh) [inline guides/plugins/apps/gateways/checkout/checkout-gateway.md:47]: Use `::: warning` instead ([docs](https://vitepress.dev/guide/markdown#default-title))

File: guides/plugins/apps/gateways/checkout/checkout-gateway.md [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+**Connection timeouts**
+
+The Shopware shop will wait for a response for 5 seconds. Be sure, that your checkout gateway implementation on your app server responds in time, otherwise Shopware will time out and drop the connection.
+{% endhint %}
+
+Your app server can then respond with a list of commands to manipulate the cart, payment methods, shipping methods, or add cart errors.
+Let's assume, your payment method is not available for carts with a total price above 1000€.
+
+{% tabs %}
```
Comment (bojanrajh) [inline guides/plugins/apps/gateways/checkout/checkout-gateway.md:56]: Use `<Tabs>` component instead ([example](https://github.com/shopware/docs/blob/3049f905a176b861b3c135490e46885e0a21f088/guides/installation/devenv.md?plain=1#L26)).

**Review by bojanrajh — CHANGES_REQUESTED** [review]

**Review by mstegmeyer — APPROVED** [review]

**Review by bojanrajh — APPROVED** [review]

**Pull request merged**